scipy/stats/_continuous_distns.py

```diff
@@ -6121,19 +6122,82 @@ def _entropy(self, c):
     def fit(self, data, *args, **kwds):
         parameters = _check_fit_input_parameters(self, data, args, kwds)
         data, fshape, floc, fscale = parameters
-        if floc is None:
-            return super().fit(data, **kwds)
+
+        ndata = data.shape[0]
+
+        def get_shape(scale, location):
+            # The partial derivative w.r.t. the shape can be solved in
+            # closed form for the shape.
+            return ndata / np.sum(np.log((data - location) / scale))
+
+        if floc is fscale is None:
+            # Optimization over a system of equations is possible when the
+            # location of the data is not set. The optimization requires that
+            # the scale be free, but the shape may be fixed. The equations
+            # are from the partial derivatives of the log-likelihood function.
```

Comment:
What does "Optimization over a system of equations" mean? We're solving a system of equations.

That said, I don't think we can explain everything concisely here. We'll need to refer to the math written elsewhere.

```suggestion
            # The support of the distribution is (x - loc)/scale > 0
            # The method of Lagrange multipliers turns this constraint
            # into an equation that can be solved numerically.
            # See gh-12545 for details.
```